src/chain/tracker.rs

```diff
@@ -324,30 +324,30 @@ impl ChainWatcher {
             &chain.asset
         );
         // Remove unwanted assets
-        assets.retain(|asset_name, properties| {
-            tracing::info!(
-                "chain {} has token {} with properties {:?}",
-                &chain.name,
-                asset_name,
-                properties
-            );
+        assets = assets
+            .into_iter()
+            .filter_map(|(asset_name, properties)| {
+                tracing::info!(
+                    "chain {} has token {} with properties {:?}",
+                    &chain.name,
+                    &asset_name,
+                    &properties
+                );
 
-            if let Some(ref native_token) = chain.native_token {
-                (native_token.name == *asset_name) && (native_token.decimals == specs.decimals)
-            } else {
                 chain
                     .asset
                     .iter()
-                    .any(|a| a.name == *asset_name && Some(a.id) == properties.asset_id)
-            }
-        });
+                    .find(|a| Some(a.id) == properties.asset_id)
+                    .map(|a| (a.name.clone(), properties))
+            })
+            .collect();
 
-        if let Some(ref native_token) = chain.native_token {
+        if let Some(native_token) = chain.native_token.clone() {
             if native_token.decimals == specs.decimals {
                 assets.insert(
-                    native_token.name.clone(),
+                    native_token.name,
```

Comment:
I just copied my fixes from #116. It does the same as I see.

Reply:
this just clones the whole native token every time, instead of just name in some cases, just makes things slightly worse? Maybe remove it since we've noticed it?

Reply:
Addressed in https://github.com/Alzymologist/Kalatori-backend/pull/107/commits/6730f5d340ce3ee03b52a65ca79e90fa8c4afea8.